src/platform/tls_openssl.c

```diff
@@ -2127,18 +2134,46 @@ QuicHpComputeMask(
     _Out_writes_bytes_(QUIC_HP_SAMPLE_LENGTH * BatchSize) uint8_t* Mask
     )
 {
-    for (uint8_t i = 0; i < BatchSize; i++) {
-        if (!QuicTlsHeaderMask(
-                Mask + i * QUIC_HP_SAMPLE_LENGTH,
-                Key->Buffer,
-                Key->BufferLen,
-                Cipher + i * QUIC_HP_SAMPLE_LENGTH,
-                Key->Aead)) {
-            return QUIC_STATUS_TLS_ERROR;
+    BOOLEAN Ret = FALSE;
+    int Len = 0;
+    uint32_t Offset = 0;
+    static const uint8_t PLAINTEXT[] = "\x00\x00\x00\x00\x00";
+
+    for (uint8_t i = 0; i < BatchSize; ++i) { // TODO - Figure out how to not use a loop here!
+        if (EVP_EncryptInit_ex(Key->CipherCtx, Key->Aead, NULL, Key->Buffer, Cipher + Offset) != 1) {
+            QuicTraceEvent(
+                LibraryError,
+                "[ lib] ERROR, %s.",
+                "EVP_EncryptInit_ex failed");
+            goto Exit;
         }
+
+        if (EVP_EncryptUpdate(Key->CipherCtx, Mask + Offset, &Len, PLAINTEXT, sizeof(PLAINTEXT) - 1) != 1) {
```

Comment:
If you make the above suggestion, you can remove the `- 1` for the terminating NULL here.